apps/client/src/app/modules/list/data/ListRowSerializationHelper.ts

```diff
@@ -0,0 +1,315 @@
+    return {
+      ...obj,
+      npcs: this.serializeNPCs(obj.npcs),
+      trades: obj.trades ? obj.trades.map(t => this.serializeTrade(t)) : undefined,
+    };
+  }
+
+  public applyMonsterInfo(obj: MonsterInfo): MonsterInfoWithMobName {
+    const id = obj.id ? obj.id : obj.itemId ? obj.itemId : undefined;
+    const mobName = this.getMobNameFromId(id);
+    const retval = {
+      ...obj,
+      name: this.getNameIfExists(mobName),
+      zoneName: this.getZoneName(obj.position ? obj.position : obj),
+    };
+    return retval;
+  }
+
+  private getMobNameFromId(id: number): I18nName {
+    return this.l12n.getMob(MobNamePipe.getActualMobId(id));
+  }
+
+  private getMonsterHuntData(monsterDrops: MonsterInfo[]): MonsterInfoWithMobName[] {
+    return monsterDrops && monsterDrops.length > 0 ? monsterDrops.map((r: any) => this.applyMonsterInfo(r)) : undefined;
+  }
+
+  public serializeNPCs(vendors: Vendor[]): VendorWithName[] {
+    return vendors && vendors.length > 0 ? vendors.map((r: Vendor) => this.applyNpcName(r)) : undefined;
+  }
+
+  private serializeVentures(item: ListRow, ventures: number[]) {
+    return ventures && ventures.length > 0 ? this.gt.getVentures(ventures).map(venture => {
+      const retval = {
+        ...venture,
+        ventureDetails: VenturesComponent.ventureAmounts(venture)
+          .map(threshold => {
+            return {
+              ...threshold,
+              venturesRemaining: Math.ceil((item.amount - item.done) / threshold.quantity)
+            }
+          }),
+        name: this.getNameIfExists(this.l12n.getVenture(venture.id)),
+        job: this.getJobAbbreviationFromId(venture.job),
+      };
+      delete retval.gathering;
+      delete retval.amounts;
+      delete retval.ilvl;
+      delete retval.jobs;
+      return retval;
+    }) : undefined;
+  }
+
+  private applyIntanceName(instance: Instance): Instance {
+    const retval = { ...instance }
+    if (instance != null && !instance.name)
+      retval.name = this.getNameIfExists(this.l12n.getInstanceName(instance.id))
+    return retval;
+  }
+
+  private serializeInstances(instances: Instance[]): InstanceWithName[] {
+    return instances && instances.length > 0 ? instances.map((r: Instance) => this.applyIntanceName(r)) : undefined;
+  }
+
+  private serializeReducedFrom(reducedFrom: ItemNameHolder[]): ItemNameHolder[] {
+    return reducedFrom && reducedFrom.length > 0 ? reducedFrom.map((r: any) => this.applyItemName(r)) : undefined;
+  }
+
+  private serializeTreasures(treasures: ItemNameHolder[]): ItemNameHolder[] {
+    return treasures && treasures.length > 0 ? treasures.map((r: any) => this.applyItemName(r)) : undefined;
+  }
+
+  private serializeMasterbooks(masterbooks: ItemNameHolder[]): ItemNameHolder[] {
+    return masterbooks && masterbooks.length > 0 ? masterbooks.map((r: any) => this.applyItemName(r)) : undefined;
+  }
+
+  private serializeGardening(gardening: any) {
+    if (!Number.isInteger(gardening))
```

Comment:
If you don't accept anything else than number, type the input to `number` instead of `any` plus this code logic.